Homework-06-documentation-testsanswers.ipynb

```diff
@@ -0,0 +1,201 @@
+    "\n",
+    "stat = TextStats(text)\n",
+    "print(stat.count_words())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Assignment 3\n",
+    "\n",
+    "Start working on your project. Create a GitHub repository for your code. Commit and push your work. You should have at least started on your code, so that we can see an overall plan for your project. Send Ken and Julia the link to your repository."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "SyntaxError",
+     "evalue": "invalid syntax (<ipython-input-4-3f0651a67d3d>, line 37)",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-4-3f0651a67d3d>\"\u001b[0;36m, line \u001b[0;32m37\u001b[0m\n\u001b[0;31m    with open('file','r')\u001b[0m\n\u001b[0m                         ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
+     ]
+    }
+   ],
+   "source": [
+    "# should be in separate repository.\n",
+    "import os\n",
+    "\n",
+    "for path, dirs, files in os.walk ('/Users/hinmingfrankiechik/Desktop/text_files'):\n",
+    "    for file in files:\n",
+    "        print(os.path.join(path, file))\n",
+    "    \n",
+    "\n",
+    "with os.scandir('/Users/hinmingfrankiechik/Desktop/text_files') as dircontent:\n",
+    "    for name in dircontent:\n",
+    "        print(name)\n",
+    "with  os.scandir('/Users/hinmingfrankiechik/Desktop/text_files/2001') as dircontent:\n",
+    "    for name in dircontent:\n",
+    "        print(name)\n",
+    "with  os.scandir('/Users/hinmingfrankiechik/Desktop/text_files/2002') as dircontent:\n",
+    "    for name in dircontent:\n",
+    "        print(name)\n",
+    "with  os.scandir('/Users/hinmingfrankiechik/Desktop/text_files/2003') as dircontent:\n",
+    "    for name in dircontent:\n",
+    "        print(name)\n",
+    "with  os.scandir('/Users/hinmingfrankiechik/Desktop/text_files/2004') as dircontent:\n",
+    "    for name in dircontent:\n",
+    "        print(name)\n",
+    "with  os.scandir('/Users/hinmingfrankiechik/Desktop/text_files/2005') as dircontent:\n",
+    "    for name in dircontent:\n",
+    "        print(name)\n",
+    "with  os.scandir('/Users/hinmingfrankiechik/Desktop/text_files/2007') as dircontent:\n",
+    "    for name in dircontent:\n",
+    "        print(name)\n",
+    "with  os.scandir('/Users/hinmingfrankiechik/Desktop/text_files/2008') as dircontent:\n",
+    "    for name in dircontent:\n",
+    "        print(name)\n",
+    "with  os.scandir('/Users/hinmingfrankiechik/Desktop/text_files/2009') as dircontent:\n",
+    "    for name in dircontent:\n",
+    "        print(name)\n",
```

Comment:
all of the above statements are not needed. Your first `os.walk` call gives you all files in all subfolders.